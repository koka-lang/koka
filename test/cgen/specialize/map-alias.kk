fun map2(xs : list<_>, f) {
    map(xs, f)
}

fun map3(xs : list<_>, f) {
    map2(xs, f)
}

fun main() {
    [1,2,3].map3(fn(x) { x + 1 }).show.println
}