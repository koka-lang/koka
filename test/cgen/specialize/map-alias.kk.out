[2,3,4]
 
cgen/specialize/map-alias/.lift000-main: (xs : list<int>) -> console list<int>
cgen/specialize/map-alias/main: () -> console ()
cgen/specialize/map-alias/map2: forall<a,b,e> (xs : list<a>, f : (a) -> e b) -> e list<b>
cgen/specialize/map-alias/map3: forall<a,b,e> (xs : list<a>, f : (a) -> e b) -> e list<b>